New navbar layout

suryatho opened the pull request:
## Changes

Move around items on the nav bar, and remove some items to put inside extra options menu.

## Notes

Questionable use cases of ng::deep in certain areas to color the icons properly. Changed method of finding the `selectedRoute` in `app-nav-bar.component.ts` to subscribe for any changes using the router service. This was because there was a problem when selecting a different tab in the menu dialog, the selected tab in the main nav bar did not update. 

## Screenshots

_Any remaining things that need to get done_
<img width="1512" height="675" alt="image" src="https://github.com/user-attachments/assets/32ad0d93-f93c-4075-aa8f-7c94f468af70" />

<img width="1512" height="543" alt="image" src="https://github.com/user-attachments/assets/94a8b941-0976-452f-87c2-32075f7dee1b" />

<img width="1511" height="738" alt="image" src="https://github.com/user-attachments/assets/a5de07e4-f8e6-499d-af24-19847f1acd24" />

Smaller width
<img width="921" height="860" alt="image" src="https://github.com/user-attachments/assets/d7b3e9ec-0325-4ef0-ae78-72c8f342c27c" />

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `package-lock.json` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #436

suryatho committed:
- Use logo for home button
- Move around items on nav bar
- Don't show messages per second if none

suryatho committed:
Add all changes:
- Update main nav bar to correct order
- Add menu with dialog (questionable ng::deep)
- Address problem where selecting option in the menu would keep item in main nav bar as still "selected" (red)

suryatho committed:
Fix formatting

bracyw commented on angular-client/src/app/app-nav-bar/app-nav-bar.component.css:
```diff
@@ -54,6 +72,46 @@
+
+/* Target the SVG inside mat-icon */
+.navbar-menu:hover mat-icon ::ng-deep svg,
+.navbar-menu.selected mat-icon ::ng-deep svg {
```
I was wrong when we first spoke. I think this can be fixed by #450  , so just review that and merge it to develop. Pull develop locally. Then merge into you're branch and use it to download the new icons you used. Take a look @ app-context.component.ts and import new svg's like that (if they aren't already)

bracyw replied:
we can talk about @ meeting tmrw will be easy fix.

bracyw commented on angular-client/src/app/app-nav-bar/app-nav-bar.component.html:
```diff
@@ -29,27 +32,26 @@
+        popUpSeverity="success"
+        popUpTitle="New Run Started"
+        popUpDetails="Your new run has started successfully."
+        icon="360"
```
where is this icon actually used, not sure if it's being applied anywhere.

bracyw commented on angular-client/src/app/app-nav-bar/app-nav-bar.component.html:
```diff
@@ -29,27 +32,26 @@
-    </div>
+      <messages-per-second />
+      <div class="navbar-menu" (click)="toggleMenu()" [class.selected]="menuRef !== undefined">
+        <mat-icon [svgIcon]="'menu'" />
```
you can probably use this icon to match figma better (download with new script mentioned above): https://fonts.google.com/icons?selected=Material+Symbols+Outlined:apps:FILL@0;wght@400;GRAD@0;opsz@24&icon.size=24&icon.color=%23e8eaed

bracyw commented on angular-client/src/components/nav-options-menu/nav-options-menu.component.ts:
```diff
@@ -0,0 +1,68 @@
+    });
+  };
+
+  menuItems: MenuItem[] = [
```
maybe pass this as input to the component. Feel free to disagree, just thinking definition in nav-bar would be good, then we have like an auto overflow into this component.

bracyw commented on angular-client/src/components/nav-options-menu/nav-options-menu.component.css:
```diff
@@ -0,0 +1,44 @@
+}
+
+.menu-icon {
+  font-size: 48px;
```
Can you also do this to main nav bar items. It looks nice.

bracyw replied:
maybe not quite that big but still main nav bar could get bigger (I never got around to it)

bracyw commented on angular-client/src/components/nav-options-menu/nav-options-menu.component.html:
```diff
@@ -0,0 +1,8 @@
+<div class="menu-container">
+  <div class="menu-grid">
+    <div *ngFor="let item of menuItems" class="menu-item" (click)="item.onclick()" [attr.data-id]="item.id">
```
use `@for` (see in other components) new for angular 19

bracyw replied:
instead of ngFor

bracyw replied:
also for using AI obviously sparingly and check output. But Angular actually supplies prompts to try and push agents / chatbots to output good code: https://angular.dev/ai

bracyw replied:
I haven't used yet but probably will and it would be interesting if you saw better output.

bracyw reviewed (changes requested):
Couple changes. Looking good, I like the sizing for the left side of nav bar. Feel free to just ask questions @ meeting tmrw, it's really not that much just hard to type out.

suryatho committed:
Merge remote-tracking branch 'origin/develop' into 436-new-navbar-layout

suryatho committed:
Address comments on pr
- stop using ::ng-deep
- make items on nav bar bigger
- use google icons for edit run button and menu button
- added rule in typography.component.css to inherit color because

bracyw reviewed (changes requested):
Looks good, one last change + formatting.

Can you make it so that overflow works for smaller screen widths on nav bar.

<img width="2880" height="1590" alt="Image" src="https://github.com/user-attachments/assets/4b9cae7f-cb55-47b1-b201-e59da0711b31" />

1000px width working would be good for this pr:
two options:
1. have icon overflow + start new run get's thrown into the app icon
2. (probably better / easier): shrink text @ certain screen width + use 24 px icons

Once that's done we're all good and will merge. You can probably just do this during Thursday meeting.

suryatho committed:
Make nav bar text smaller when window is small.
Update typography component to not use the @Input annotation
#436
